src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,45 +3,150 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private Node<T> head;
+    private Node<T> tail;
+    private int size;
+
+    private static class Node<T> {
```

Comment:
Make `Node<T>` an inner class. The purpose of inner classes is to group classes that belong together, which makes your code more readable and maintainable. Inner class should be at the end of the class.